docs/source/tutorial_worlds.md

```diff
@@ -13,120 +13,353 @@ section. If you are interested in creating a task, please see
+            self.model = shared['model']
+
+    def share(self):
+        # This is used to create the "shared" dictionary that clones will
+        # receive in initialization. Make sure you enumerate everything
+        # that needs to be seen in all clones!
+        shared = super().share()
+        shared['model'] = self.model
+        return shared
+```
 
+:::{note} Real implementations
+The above is really just a very simple example. You may find it illuminating to
+read the implementation of
+[UnigramAgent](https://github.com/facebookresearch/ParlAI/blob/master/parlai/agents/unigram/unigram.py)
+or even [TorchAgent](parlai.core.torch_agent.TorchAgent).
+:::
+
+
+:::{note} Usefulness
+The above pseudo-code is likely to be used frequently as a user of ParlAI.
+Some things are automatically shared by our abstractions (like models and
+optimizers), but some users may have to include extra sharing.
+:::
+
+Each clone of a model is relatively cheap to create: the only new memory used
+is the dialogue context. The share object enables us to reuse memory for
+expensive objects, like neural network weights.
+
+At a glance, creating these clones may seem strange. Why do we create all these
+copies? Differentiating between shared and independent state lets us write our
+agents with _independence_ in mind. Agents only need to __focus on a single
+conversation__, maintaining and manipulating the state for just that conversation.
+Every clone will be maintaining _separate conversations_, so when we implement
+a new dataset or model, we only focus on one conversation at a time.
+
+This is what makes up the fundamental backend behind batching or chat services:
+each clone of the agent only has to focus on one conversation at a time, with
+only specific spots for synchronization. In the next section, we'll
+take a look at how this is used in Batching.
+
+## Batching
+
+:::{warning} Already implemented in `TorchAgent`.
+If you're implementing your own custom Neural Network model, you don't need to
+worry about this, except at a high level. This is all handled for you by
+`TorchAgent`.
+:::
+
+Equipped with the ability to create clones, we can use this to implement
+Batching. For a batchsize of 3, we will create 3 clones of the Teacher, and 3
+clones of the Agent. Each of these clones will maintain their own separate,
+independent conversations. Naively, this could be implemented with a simple
+for loop:
+
+```python
+class NaiveBatchWorld(World):
+    def __init__(self, opt, agents):
+        # store the originals
+        self.teacher, self.model = agents
+        # make batchsize copies of all the models
+        self.teacher_copies = []
+        self.model_copies = []
+        self.batchsize = opt['batchsize']
+        for i in range(self.batchsize):
+            self.teacher_copies.append(self.teacher.clone())
+            self.model_copies.append(self.model.clone())
+
+```
+This initialization code is then represented by this graphic:
```

Comment:
Is agent0 the manager (master) here and the rest its workers? Does agent0 itself run the conversations, or is it just handling the clones in the batch?